### PR TITLE
Extend functional test for verifying switch rules synchronisation

### LINKED ISSUE #1644: Test rules synchronization on transit switches
Extend existing `Able to synchronize rules on a switch` test to test ingress/egress/transit switch rules cases

### LINKED ISSUE #1842: Add more synch rules tests
Take into account #1662
Add more tests or verifications that explicitly synchronize ingress/egress and transit rules.

### PR DESCRIPTION
**Pull request by ylobankov**
Closes #1842.
Closes #1644.

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy [hunk 1]
```diff
@@ -387,33 +387,53 @@ class SwitchRulesSpec extends BaseSpecification {
...
+
+        and: "Create a transit-switch flow going through these switches"
         def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
         flowHelper.addFlow(flow)
 
-        def defaultPlusFlowRules = northbound.getSwitchRules(srcSwitch.dpId).flowEntries
-        northbound.deleteSwitchRules(srcSwitch.dpId, DeleteRulesAction.IGNORE_DEFAULTS)
-        Wrappers.wait(RULES_DELETION_TIME) {
-            assert northbound.getSwitchRules(srcSwitch.dpId).flowEntries.size() == srcSwDefaultRules.size()
+        and: "Reproduce situation when switches have missing rules by deleting flows rules from them"
```
Comment (npliashechnikov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy]: Please change flows rules to flow rules

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy [hunk 2]
```diff
@@ -387,33 +387,53 @@ class SwitchRulesSpec extends BaseSpecification {
...
         and: "No missing rules were found after rules validation"
-        with(northbound.validateSwitchRules(srcSwitch.dpId)) {
-            verifyAll {
-                properRules.size() == flowRulesCount
-                missingRules.empty
-                excessRules.empty
+        involvedSwitches.each { switchId ->
+            with(northbound.validateSwitchRules(switchId)) {
+                verifyAll {
+                    properRules.size() == flowRulesCount
```
Comment (npliashechnikov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy]: I'd also check if the rule cookies are the same as before deleting them.

**Comment by sergii-iakovenko** [comment]
@ylobankov Please, resolve all review conversations.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy (hunk 1 above)
Comment (ylobankov) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy]: done.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy (hunk 2 above)
Comment (ylobankov) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy]: It is checked at L425. Moreover, not only cookies are checked there.

**Pull request merged**